lib/config/ChainConfigManager.ts

```diff
@@ -0,0 +1,300 @@
+import { ChainId } from '@uniswap/sdk-core';
+import { RoutingType } from '../constants';
+
+type CommonConfig = {
+  routingType: RoutingType;
+  priceImprovementBps?: number;
+  stdAuctionPeriodSecs?: number;
+  deadlineBufferSecs?: number;
+}
+
+export type DutchConfig = CommonConfig & {
+  routingType: RoutingType.DUTCH_LIMIT | RoutingType.DUTCH_V2;
+  skipRFQ?: boolean;
+}
+
+type ChainConfig = {
+  routingTypes: (
+    | CommonConfig & {
+        routingType: RoutingType.CLASSIC | RoutingType.RELAY;
+      }
+    | DutchConfig & {
+        routingType: RoutingType.DUTCH_V2;
+      }
+    | DutchConfig & {
+        routingType: RoutingType.DUTCH_LIMIT;
+        largeAuctionPeriodSecs?: number;
+      }
+  )[];
+  alarmEnabled: boolean;
+};
+
+export abstract class ChainConfigManager {
+  // Represents the other route dependencies for each route type
+  // If a route is added in to the supported routingTypes for a chain,
+  // all dependencies will also be supported
+  private static readonly _routeDependencies: { [routingType: string]: RoutingType[] } = {
+    [RoutingType.DUTCH_LIMIT]: [RoutingType.CLASSIC],
+    [RoutingType.DUTCH_V2]: [RoutingType.CLASSIC],
+  };
+  private static readonly _chainConfigs: { [chainId: number]: ChainConfig } = {
+    [ChainId.MAINNET]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC,
+        },
+        {
+          routingType: RoutingType.DUTCH_LIMIT,
+          largeAuctionPeriodSecs: 120,
+        },
+        {
+          routingType: RoutingType.RELAY,
+        },
+        {
+          routingType: RoutingType.DUTCH_V2,
+          // 25 blocks from now
+          // to cover time to sign, run secondary auction, and some blocks for decay
+          deadlineBufferSecs: 300,
+        },
+      ],
+      alarmEnabled: true,
+    },
+    [ChainId.OPTIMISM]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC,
+        },
+      ],
+      alarmEnabled: true,
+    },
+    [ChainId.OPTIMISM_GOERLI]: {
+      // TODO: add back optimism GOERLI once we are sure routing api supports it
+      routingTypes: [],
+      alarmEnabled: false,
+    },
+    [ChainId.ARBITRUM_ONE]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC,
+        },
+        {
+          routingType: RoutingType.DUTCH_V2,
+          skipRFQ: true,
+          priceImprovementBps: 2,
+        },
+      ],
+      alarmEnabled: true,
+    },
+    [ChainId.ARBITRUM_GOERLI]: {
+      routingTypes: [
```

Comment:
one idea for slightly more concise config def is to have default config creator fns like:
```
function classicConfig(overrides: Partial<ClassicConfig>): ClassicConfig {
  return { routingType: RoutingType.CLASSIC, ...overrides };
}

then each standard inclusion here is only one line

routingTypes: [
  classicConfig(), 
  dutchConfig({ priceImprovementBps: 1 })
]
```

Reply:
I've done some refactoring in the definition which gives a uniqueness check. LMK what you think about the new method.